lib/spotlight/dor/indexer.rb

```diff
@@ -50,6 +52,12 @@ def add_box(sdb, solr_doc)
         solr_doc['box_ssi'] = sdb.smods_rec.box
       end
 
+      # pulls from //subject/geographic
+      def add_geonames(sdb, solr_doc)
+        ids = extract_geonames_ids(sdb)
+        solr_doc['geographic_srpt'] = ids.map { |id| get_geonames_api_envelope(id) }.reject(&:nil?)
```

Comment:
FWIW `compact` works [down at the C layer](https://github.com/ruby/ruby/blob/f14d64638a461ff660d16b42ae827b46a0cec91b/array.c#L4412), so it is more efficient:

```
Benchmark.bm do |x|
  x.report('Compact:') { array.compact }
  x.report('Reject nil:') { array.reject(&:nil?) }
end
          user     system      total        real
Compact:     0.000000   0.000000   0.000000 (  0.000008)
Reject nil:  0.000000   0.000000   0.000000 (  0.000102)
```